updates:

wyq24 opened the pull request:
1: add 'ThermalDistributionMinimizerOneSrc' as a new cost function
2: add 'Paste Result to Init' to 'Parameters' in the menu bar which allows user to set the current result to the initial value of each parameter.

wyq24 committed:
updates:
1: add 'ThermalDistributionMinimizerOneSrc' as a new cost function
2: add 'Paste Result to Init' to 'Parameters' in the menu bar which allows user to set the current result to the initial value of each parameter.

binchensun commented on utils/gstools.py:
```diff
@@ -289,3 +297,5 @@ def SinglePowerLawMinimizerOneSrc(fit_params, freqghz, spec=None, spec_err=None,
             return (mflux - spec) / spec_err
 
+    #Define thermal_distribution case as a new function.
+    ThermalDistributionMinimizerOneSrc = functools.partial(SinglePowerLawMinimizerOneSrc, elec_distribution=2)
```
Good idea!

binchensun commented on pygsfit.py:
```diff
@@ -1846,6 +1855,25 @@ def init_params(self):
             self.fit_function = gstools.GSCostFunctions.Ff_Gyroresonance_MinimizerOneSrc
             self.update_fit_param_widgets()
 
+        if self.ele_dist == 'thermal':
```
I already have a hook to "thermal f-f + gyrores" (lines 1839-1858). Let us merge the two blocks. When setting parmlocal[6]=2 in fast GS code, it uses thermal f-f + gyroresonance. I wish to distinguish this from pure free-free (a cost function needs to be written, but pretty straightforward).

binchensun reviewed (changes requested):
Please merge two blocks as indicated in my comment (thermal -> thermal f-f + gyrores) and submit another PR.

wyq24 committed:
updates:
1: add 'ThermalDistributionMinimizerOneSrc' as a new cost function
2: add 'Paste Result to Init' to 'Parameters' in the menu bar which allows user to set the current result to the initial value of each parameter.

wyq24 commented:
updates:
1: add 'ThermalDistributionMinimizerOneSrc' as a new cost function
2: add 'Paste Result to Init' to 'Parameters' in the menu bar which allows user to set the current result to the initial value of each parameter.

wyq24 committed:
add flux factor of 20220511

wyq24 committed:
new factors
